modules/ding_availability/ding_availability.field.inc

```diff
@@ -225,8 +225,11 @@ function ding_availability_field_formatter_view($entity_type, $entity, $field, $
             $url = entity_uri('ting_collection', $entities[0]);
           }
 
-          // Mark as available if it is an online resource.
-          if (in_array(drupal_strtolower($type), $online_types) || $entities[0]->getOnline_url(FALSE)) {
+          // Mark as available if it is an online resource. As some types are
+          // not always predefined online type we need to check if the entity
+          // has an online URL. Please not tht that this in most cases will
+          // trigger a search request with get all relations.
+          if (in_array(drupal_strtolower($type), $online_types) || $entities[0]->getOnline_url()) {
```

Comment:
Given the change to `opensearch_execute` additional search requests should be prevented.

Reply:
They should but in some cases with list and cold cache this might still trigger an extra search request or get object.